### PR TITLE
DRY code

### PR DESCRIPTION
**Pull request by ligi**
+43 -100 ;-)

### CODE REVIEW

File: Library/src/main/java/info/metadude/android/typedpreferences/BasePreference.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package info.metadude.android.typedpreferences;
+
+import android.content.SharedPreferences;
+
+/**
+ * A wrapper class for a boolean preference.
```
Comment (johnjohndoe) [inline Library/src/main/java/info/metadude/android/typedpreferences/BasePreference.java]: Please clean up the comments, `An abstract base class for preference.` might be fitting.

File: Library/src/main/java/info/metadude/android/typedpreferences/BasePreference.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ */
+class BasePreference {
```
Comment (johnjohndoe) [inline Library/src/main/java/info/metadude/android/typedpreferences/BasePreference.java]: To keep the same style as the other classes please add `public`. The class also might be `abstract`, do you agree?

Comment (ligi) [inline Library/src/main/java/info/metadude/android/typedpreferences/BasePreference.java]: no this should not be public as it should not be use outside this package - also not abstract as it has no abstract methods

File: Library/src/main/java/info/metadude/android/typedpreferences/BasePreference.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
+package info.metadude.android.typedpreferences;
+
+import android.content.SharedPreferences;
+
+class BasePreference {
+
+    protected final SharedPreferences mPreferences;
+    protected final String mKey;
+
+    public BasePreference(final SharedPreferences preferences, final String key) {
```
Comment (johnjohndoe) [inline Library/src/main/java/info/metadude/android/typedpreferences/BasePreference.java]: How do you suggest the two parameters constructor can be hidden? It is currently accessible from the test classes.

Comment (ligi) [inline Library/src/main/java/info/metadude/android/typedpreferences/BasePreference.java]: ? before there was also a 2 parameter constructor - I see no change

Comment (johnjohndoe) [inline Library/src/main/java/info/metadude/android/typedpreferences/BasePreference.java]: True. Sorry, was a little late yesterday.

**Comment by johnjohndoe** [comment]
Merged. Thanks for the improvement!

**Pull request merged**